### PR TITLE
Header tests

### PR DESCRIPTION
**Pull request by Rombur**
Add tests to check that we are not missing headers (spoiler: we are missing some). This creates a lot of new tests but we just compile them, we don't run them.

relates to #7

### CODE REVIEW

**Comment by aprokop** [comment]
Request per @dalg24: DO NOT MERGE YET

**Comment by dalg24** [comment]
@Rombur please review

File: test/headers_self_contained/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Create tests that contain each header seperately. We do not not run these test
```
Comment (Rombur) [inline test/headers_self_contained/CMakeLists.txt]: There are two typos here (my bad), should be `We do not run these tests`

Comment (Rombur) [inline test/headers_self_contained/CMakeLists.txt]:
```suggestion
# Create tests that contain each header separately. We do not  run these tests
```

Comment (dalg24) [reply test/headers_self_contained/CMakeLists.txt]: `You do not have permission to push to this repository. Refresh and try again.`??!

Comment (dalg24) [reply test/headers_self_contained/CMakeLists.txt]: This makes no sense I pushed to it earlier

Comment (dalg24) [reply test/headers_self_contained/CMakeLists.txt]: Or was it because it was on a branch associated to a PR

**Pull request merged**